lightningd/channel_control.c

```diff
@@ -2079,6 +2104,122 @@ static const struct json_command splice_signed_command = {
 };
 AUTODATA(json_command, &splice_signed_command);
 
+static struct command_result *json_splice_out(struct command *cmd,
+					     const char *buffer,
+					     const jsmntok_t *obj UNNEEDED,
+					     const jsmntok_t *params)
+{
+	u8 *msg;
+	struct splice_command *cc;
+	struct channel *channel;
+	struct amount_sat *amount;
+	s64 relative_amount;
+	u32 *feerate_per_kw;
+	bool *force_feerate;
+	struct wally_psbt *psbt;
+	u32 *locktime;
+	struct pubkey pubkey;
+	s64 keyidx;
+	u8 *b32script;
+	size_t weight;
+	bool *sign_first;
+
+	if (!param(cmd, buffer, params,
+		  p_req("channel_id", param_channel_for_splice, &channel),
+		  p_req("amount", param_sat_or_all, &amount),
+		  p_opt("feerate_per_kw", param_feerate, &feerate_per_kw),
+		  p_opt_def("force_feerate", param_bool, &force_feerate, false),
+		  p_opt("initialpsbt", param_psbt, &psbt),
+		  p_opt("locktime", param_number, &locktime),
+		  p_opt_def("sign_first", param_bool, &sign_first, false),
+		  NULL))
+		return command_param_failed();
+
+	if (!feerate_per_kw) {
+		feerate_per_kw = tal(cmd, u32);
+		*feerate_per_kw = opening_feerate(cmd->ld->topology);
+	}
+
+	if (!psbt) {
+		if (!locktime) {
+			locktime = tal(cmd, u32);
+			*locktime = default_locktime(cmd->ld->topology);
+		}
+		psbt = create_psbt(cmd, 0, 0, *locktime);
+	}
+	else if(locktime) {
+		return command_fail(cmd, FUNDING_PSBT_INVALID,
+				    "Can't set locktime of an existing"
+				    " {initialpsbt}");
+	}
+
+	if (!validate_psbt(psbt))
+		return command_fail(cmd,
+				    FUNDING_PSBT_INVALID,
+				    "PSBT failed to validate.");
+
+	if (amount_sat_less(*amount, chainparams->dust_limit))
+		return command_fail(cmd, FUND_OUTPUT_IS_DUST,
+				    "amount is below dust limit (%s)",
+				    type_to_string(tmpctx,
+				    		   struct amount_sat,
+				    		   &chainparams->dust_limit));
+
+	if (splice_command_for_chan(cmd->ld, channel))
+		return command_fail(cmd,
+				    SPLICE_BUSY_ERROR,
+				    "Currently waiting on previous splice"
+				    " command to finish.");
+
+	/* Get a change adddress */
+	keyidx = wallet_get_newindex(cmd->ld);
+	if (keyidx < 0)
+		return command_fail(cmd, LIGHTNINGD,
+				    "Failed to generate change address."
+				    " Keys exhausted.");
+
+	b32script = NULL;
+	if (chainparams->is_elements) {
+		bip32_pubkey(cmd->ld, &pubkey, keyidx);
+		b32script = scriptpubkey_p2wpkh(tmpctx, &pubkey);
+	} else {
+		b32script = p2tr_for_keyidx(tmpctx, cmd->ld, keyidx);
+	}
+	if (!b32script) {
+		return command_fail(cmd, LIGHTNINGD,
+				    "Failed to generate change address."
+				    " Keys generation failure");
+	}
+	txfilter_add_scriptpubkey(cmd->ld->owned_txfilter, b32script);
+
+	psbt_append_output(psbt, b32script, *amount);
+
+	weight = wally_psbt_weight(psbt);
```

Comment:
i think you're going to want to add the signature witness byte weight as well, or at least an estimate for it.